tools/driver/driver.cpp

```diff
@@ -192,6 +217,16 @@ int main(int argc_, const char **argv_) {
   int ExpandedArgc = ExpandedArgs.size();
   const char **ExpandedArgv = ExpandedArgs.data();
   PROGRAM_START(ExpandedArgc, ExpandedArgv);
+#ifdef _WIN32
+  // PROGRAM_START/InitLLVM overwrites the passed arguments with the unexpanded
+  // commandline arguments. Since we already handle getting utf-8 arguments, we
+  // overwrite the changed arguments again.
```

Comment:
This seems wrong for a few reasons:

- the Swift-driver-written response files are going to be in UTF-8, aren't they?
- the PrettyStackTraceProgram inside InitLLVM isn't going to have the expanded arguments, which we were very much trying to support
- this is now doing redundant work

Maybe we shouldn't be using InitLLVM.

Reply:
- The response files are expected to be in UTF-8. This converts the file name argument to UTF16 but the expansion is still UTF8
- This does actually register the right arguments, you get the expanded arguments in "Program Arguments:". InitLLVM overwrites the arguments you pass it, but not the ones it registers with the StackPrinter. I'm unsure why.
- Yup, it's completely redundant. Now that I think about it, it would be slightly better to make copies of the expanded arguments and pass those to InitLLVM. But alternatively, it wouldn't be difficult to have an `InitSwift` instead. Any preference?

Reply:
The less we can diverge from LLVM the better, so if "just don't use the replacements InitLLVM gives us" is the right answer, we can just make extra Argc and Argv variables that we then throw away.

Reply:
Seems reasonable to me!